packages/fluentui/react-northstar-prototypes/src/prototypes/VirtualizedStickyTree/VirtualStickyTree.tsx

```diff
@@ -0,0 +1,368 @@
+        let scrollOffset = 0;
+        while (idToToggle !== stickyItemIds[i]) {
+          scrollOffset += stickyItemPusherHeights[i];
+          i++;
+        }
+        (listRef.current as any)?.scrollTo(scrollOffset);
+      }
+
+      baseToggleItemActive(e, idToToggle);
+    },
+    [activeItemIds, baseToggleItemActive, getItemById, listRef, stickyItemIds, stickyItemPusherHeights],
+  );
+
+  const contextValue: TreeRenderContextValue = React.useMemo(
+    () => ({
+      getItemById,
+      registerItemRef,
+      toggleItemActive,
+      focusItemById,
+      expandSiblings,
+      toggleItemSelect: _.noop,
+    }),
+    [getItemById, registerItemRef, toggleItemActive, focusItemById, expandSiblings],
+  );
+
+  // When using keyboard, and navigate to non-sticky items, they could be hidden behind sticky headers.
+  // Scroll to make the focused non-sticky items always visible
+  const makeVisibleOnFocus = React.useCallback(
+    (id: string, level: number) => {
+      if (level === 1) {
+        return; // focused sticky items are always visible, so no need to deal with them
+      }
+
+      const isOverlappingWithSticky = (id: string) => {
+        const overlap = (rect1: DOMRect, rect2: DOMRect) =>
+          !(rect1?.bottom <= rect2?.top || rect1?.top >= rect2?.bottom);
+
+        const itemRect = getItemRef(id)?.getBoundingClientRect();
+        for (const stickyId of stickyItemIds) {
+          const stickyRect = getItemRef(stickyId)?.getBoundingClientRect();
+          if (overlap(itemRect, stickyRect)) {
+            return true;
+          }
+        }
+        return false;
+      };
+
+      if (isOverlappingWithSticky(id)) {
+        // known issue:
+        // this scroll cannot guarantee non-sticky items' visibility. It would work when sticky titles are relatively small compare to non-sticky items.
+        listRef.current.scrollToItem(visibleItemIds.indexOf(id), 'center'); // scroll to item
+      }
+    },
+    [getItemRef, listRef, stickyItemIds, visibleItemIds],
+  );
+
+  // When using keyboard, and navigate to stickyItems, arrow up/down should navigate to previous item's last child/current Item's first child.
+  // But because of virtualization, the destination item is not always rendered, so we scroll to them to force rendering
+  const handleArrowUpDownOnSticky = React.useCallback(
+    (stickyId, stickyItem) => (e: React.KeyboardEvent) => {
+      if (e.key === 'ArrowDown') {
+        e.preventDefault();
+        focusItemById(stickyItem.childrenIds[0]);
+      }
+      if (e.key === 'ArrowUp' && stickyItem.index !== 1) {
+        const indexAmongVisible = visibleItemIds.indexOf(stickyId);
+        if (indexAmongVisible > 0) {
+          e.preventDefault();
+          focusItemById(visibleItemIds[indexAmongVisible - 1]);
+        }
+      }
+    },
+    [focusItemById, visibleItemIds],
+  );
+
+  const createTreeItem = React.useCallback(
+    (id, style) => {
+      const item = getItemById(id);
+      if (item) {
```

Comment:
nit: might be cleaner to invert this ? 
```suggestion
      if (!item) return null;
```

Reply:
:D actually I prefer it like this as well. going to make a change on this